src/compiler/compile/Component.ts

```diff
@@ -267,6 +267,14 @@ export default class Component {
 							} else {
 								let name = node.name.slice(1);
 
+								if (compile_options.hydratable) {
+									if (internal_exports.has(`${name}_hydration`)) {
+										name += '_hydration';
+									} else if (internal_exports.has(`${name}Hydration`)) {
```

Comment:
Question for understanding: What's this about? There's only methods ending with `_hydration`

Reply:
1. the function you are looking at here, is the magic that turns

```js
@insert(...)
```

into 

```js
import { insert } from 'svelte/internal';
insert(...);
```

2. `svelte/internal` exposes a lot of methods, which includes `insert` and `insert_dev`, see https://github.com/sveltejs/svelte/blob/master/src/runtime/internal/dev.ts

the function you see here, decides on whether to use `insert` or `insert_dev` from `@insert` by checking the `compileOptions.dev`, similarly, we could decide to use `insert` or `insert_hydration` based on `compileOptions.hydratable`

3. to determine the list of exports from `svelte/internal`, and then decide whether is there a `xxx_dev` exported, we build the `svelte/internal` and write them into a file `internal_exports.ts`, see

https://github.com/sveltejs/svelte/blob/931738e8219eb6bc9afe68d1c383f920c45c45cf/rollup.config.js#L86-L92

Reply:
Thanks for the thourough answer! What I don't understand though is why there's this inner `if` clause for each of these cases. The methods are called `append_hydration_dev`, but the `if` clauses look to me like there's also `appendHydrationDev` which is not the case, or am I misunderstanding something there?

Reply:
Yea.. there's `SvelteComponentDev`

https://github.com/sveltejs/svelte/blob/931738e8219eb6bc9afe68d1c383f920c45c45cf/src/runtime/internal/dev.ts#L120

and `HtmlTagHydration`

Reply:
this seems like it could be worth a code comment